dcm4che-net-audit/src/test/java/org/dcm4che3/net/audit/AuditLoggerTest.java

```diff
@@ -0,0 +1,356 @@
+    }
+
+    @Test
+    public void reconfigure_UpdatesFormatXML_WhenCalled() {
+        newLogger.setFormatXML(false);
+        logger.setFormatXML(true);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("formatXML is not as expected", logger.isFormatXML(), is(false));
+    }
+
+    @Test
+    public void reconfigure_UpdatesSupplement95_WhenCalled() {
+        newLogger.setSupplement95(true);
+        logger.setSupplement95(false);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("supplement95 is not as expected", logger.isSupplement95(), is(true));
+    }
+
+    @Test
+    public void reconfigure_UpdatesSpoolDirectory_WhenCalled() {
+        final File newDir = new File("var/local/test");
+        newLogger.setSpoolDirectory(newDir);
+        logger.setSpoolDirectory(new File(""));
+
+        logger.reconfigure(newLogger);
+
+        assertThat("spoolDirectory is not as expected", logger.getSpoolDirectory(), is(newDir));
+    }
+
+    @Test
+    public void reconfigure_UpdatesSpoolFileNamePrefix_WhenCalled() {
+        newLogger.setSpoolFileNamePrefix("PRE");
+        logger.setSpoolFileNamePrefix("POST");
+
+        logger.reconfigure(newLogger);
+
+        assertThat("spoolFileNamePrefix is not as expected", logger.getSpoolNameFilePrefix(), is("PRE"));
+    }
+
+    @Test
+    public void reconfigure_UpdatesSpoolFileNameSuffix_WhenCalled() {
+        newLogger.setSpoolFileNameSuffix("POST");
+        logger.setSpoolFileNameSuffix("PRE");
+
+        logger.reconfigure(newLogger);
+
+        assertThat("spoolFileNameSuffix is not as expected", logger.getSpoolFileNameSuffix(), is("POST"));
+    }
+
+    @Test
+    public void reconfigure_UpdatesRetryInterval_WhenCalled() {
+        newLogger.setRetryInterval(10);
+        logger.setRetryInterval(1);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("retryInterval is not as expected", logger.getRetryInterval(), is(10));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditLoggerInstalled_WhenCalled() {
+        newLogger.setAuditLoggerInstalled(false);
+        logger.setAuditLoggerInstalled(true);
+
+        logger.reconfigure(newLogger);
+
+        assertThat("auditLoggerInstalled is not as expected", logger.getAuditLoggerInstalled(), is(false));
+    }
+
+    @Test
+    public void reconfigure_UpdatesAuditRecordRepositoryDevices_WhenCalled() {
+        final Connection auditUDP = new Connection("audit-udp", "localhost", 514);
+        auditUDP.setProtocol(Connection.Protocol.SYSLOG_UDP);
+
+        final AuditRecordRepository arr1 = new AuditRecordRepository();
+        arr1.addConnection(auditUDP);
+
+        final Device arrDevice1 = new Device("syslog");
+        arrDevice1.addDeviceExtension(arr1);
+        arrDevice1.addConnection(auditUDP);
+
+        newLogger.setAuditRecordRepositoryDevices(Collections.singletonList(arrDevice1));
+
+        final Connection auditTCP = new Connection("audit-tcp", "localhost", 516);

```

Comment:
You could consider creating private helper method to create connections to reduce arrange code clutter (since it also does not matter for this tests what is actually in connections).